server/src/test/java/io/crate/auth/UserAuthenticationMethodTest.java

```diff
@@ -361,6 +375,136 @@ public void test_jwt_authentication_claim_mismatch_throws_error() throws Excepti
+        assertThat(authenticatedRole.name()).isEqualTo(tokenUsername);
+    }
+
+    @Test
+    public void test_jwt_authentication_user_has_no_jwt_properties_default_is_not_set() throws Exception {
+        String tokenUsername = "cloud_user"; // Token payload dictates CrateDB username.
+        Roles roles = () -> List.of(
+            userOf(
+                tokenUsername,
+                Set.of(),
+                new HashSet<>(),
+                getSecureHash("pwd"),
+                // User doesn't have JWT properties
+                null
+            )
+        );
+        JWTAuthenticationMethod jwtAuth = new JWTAuthenticationMethod(
+            roles,
+            Settings.EMPTY,
+            jwkProviderFunction(null),
+            () -> "test_cluster_id" // Default aud is not set, fallback to value from the token payload.
+        );
+
+        Credentials credentials = new Credentials(JWT_TOKEN);
+        credentials.setUsername(tokenUsername);
+
+        // Auth fails because empty issuer doesn't match token's issuer.
+        assertThatThrownBy(
+            () -> jwtAuth.authenticate(credentials, null))
+            .isExactlyInstanceOf(RuntimeException.class)
+            .hasMessageContaining("jwt authentication failed for user cloud_user. Reason: The Claim 'iss' value doesn't match the required issuer.");
+    }
+
+    @Test
+    public void test_jwt_authentication_user_has_no_jwt_properties_default_values_dont_match_token_payload() throws Exception {
+        String tokenUsername = "cloud_user"; // Token payload dictates CrateDB username.
+        Roles roles = () -> List.of(
+            userOf(
+                tokenUsername,
+                Set.of(),
+                new HashSet<>(),
+                getSecureHash("pwd"),
+                // User doesn't have JWT properties
+                null
+            )
+        );
+        // Defaults doesn't match JWT_TOKEN payload.
+        var settings = Settings.builder()
+            .put(AuthSettings.AUTH_HOST_BASED_JWT_ISS_SETTING.getKey(), "dummy")
+            .put(AuthSettings.AUTH_HOST_BASED_JWT_AUD_SETTING.getKey(), "dummy")
+            .build();
+        JWTAuthenticationMethod jwtAuth = new JWTAuthenticationMethod(
+            roles,
+            settings,
+            jwkProviderFunction(null),
+            () -> "test_cluster_id" // Default aud is not set, fallback to value from the token payload.
+        );
+
+        Credentials credentials = new Credentials(JWT_TOKEN);
+        credentials.setUsername(tokenUsername);
+
+        // Auth fails because default issuer doesn't match token's issuer.
+        assertThatThrownBy(
+            () -> jwtAuth.authenticate(credentials, null))
+            .isExactlyInstanceOf(RuntimeException.class)
+            .hasMessageContaining("jwt authentication failed for user cloud_user. Reason: The Claim 'iss' value doesn't match the required issuer.");
```

Comment:
Can we just create unit test ensuring the fall back logic? I do not see any value in repeating the full authentication logic just to verify the setting fallbacks are working as expected.. These new test are increasing complexity for no real reason.

Reply:
Do you mean wrapping the logic for defining iss/aud/name to some method (visible for testing) and create a single test and remove all those new tests here?

Reply:
I removed 2 tests as the are irrelevant now, leaving only 2 tests.

>Do you mean wrapping the logic for defining iss/aud/name to some method (visible for testing) and create a single test and remove all those new tests here?

Probably not worth it after last changes?

Reply:
I mean, we need one test to ensure the `JwtAuthenticationMethod` works as expected, and another to ensure the used Roles predicates work as expected. Everything else should be tested already, no need to repeat a full blown test.

Reply:
I removed this test as I added 2 tests for Predicate match. Also removed test cases in `HttpAuthUpstreamHandlerTest`